src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp

```diff
@@ -124,6 +127,13 @@ bool shared_node_optimization(const shared_ptr<Model>& model) {
                 auto& shared_nodes = item.second;
                 if (shared_nodes.size() < 2)
                     continue;
+                // sort shared_nodes so that root would be the earliest in the topological order
+                // it is critical for continuous application of this optimization
+                std::sort(shared_nodes.begin(),
```

Comment:
As I understand it,  the prev behavior could break topological sorting because we visited shared nodes in a non-topological order.
could you please add a test?

Reply:
Yes, sure